checkov/cloudformation/graph_builder/variable_rendering/renderer.py

```diff
@@ -0,0 +1,206 @@
+        super().__init__(local_graph)
+        self.evaluation_methods = {
+            IntrinsicFunctions.REF: self._evaluate_ref_connection,
+            IntrinsicFunctions.FIND_IN_MAP: self._evaluate_findinmap_connection,
+            IntrinsicFunctions.GET_ATT: self._evaluate_getatt_connection,
+            ConditionFunctions.IF: self._evaluate_if_connection,
+            IntrinsicFunctions.SUB: self._evaluate_sub_connection
+        }
+
+    def evaluate_vertex_attribute_from_edge(self, edge_list: List[Edge]) -> None:
+        multiple_edges = len(edge_list) > 1
+        edge = edge_list[0]
+        origin_vertex_attributes = self.local_graph.vertices[edge.origin].attributes
+        val_to_eval = deepcopy(origin_vertex_attributes.get(edge.label, ""))
+        vertices_block_name_map = self._extract_vertices_block_name_map()
+
+        referenced_vertices = get_referenced_vertices_in_value(
+            value=val_to_eval, vertices_block_name_map=vertices_block_name_map
+        )
+        if not referenced_vertices:
+            # DependsOn or Condition connections
+            pass
+
+        if referenced_vertices:
+            # Ref, GetAtt, FindInMap, If, Sub connections
+            cfn_evaluation_function = None
+            for curr_evaluation_function in self.EVALUATION_CFN_FUNCTIONS:
+                if curr_evaluation_function in val_to_eval:
+                    cfn_evaluation_function = curr_evaluation_function
+            if cfn_evaluation_function:
+                original_value = val_to_eval.get(cfn_evaluation_function, None)
+
+                for edge in edge_list:
+                    dest_vertex_attributes = self.local_graph.get_vertex_attributes_by_index(edge.dest)
+                    evaluated_value = self.evaluation_methods[cfn_evaluation_function](val_to_eval[cfn_evaluation_function], dest_vertex_attributes)
+                    if evaluated_value and evaluated_value != original_value:
+                        val_to_eval[cfn_evaluation_function] = evaluated_value
+                        self.update_evaluated_value(
+                            changed_attribute_key=edge.label,
+                            changed_attribute_value=evaluated_value,
+                            vertex=edge.origin,
+                            change_origin_id=edge.dest,
+                            attribute_at_dest=edge.label,
+                        )
+
+    @staticmethod
+    def _evaluate_ref_connection(value: str, dest_vertex_attributes) -> Optional[str]:
+        evaluated_value = None
+
+        # in case of Ref we take only Parameter's default value
+        if value == dest_vertex_attributes.get(CustomAttributes.BLOCK_NAME) and dest_vertex_attributes.get(
+                CustomAttributes.BLOCK_TYPE) == BlockType.PARAMETERS:
+            evaluated_value = dest_vertex_attributes.get('Default', None)
+            if evaluated_value:
+                evaluated_value = str(evaluated_value)
+
+        return evaluated_value
```

Comment:
```suggestion
        # in case of Ref we take only Parameter's default value
        evaluated_value = dest_vertex_attributes.get('Default')
        if (
            evaluated_value and
            value == dest_vertex_attributes.get(CustomAttributes.BLOCK_NAME) and 
            dest_vertex_attributes.get(CustomAttributes.BLOCK_TYPE) == BlockType.PARAMETERS
        ):
            return str(evaluated_value)

        return None
```
then you don't need the nested `if`